src/palace/manager/util/http.py

```diff
@@ -230,39 +286,24 @@ def _request_with_timeout(
         )
         backoff_factor: float = float(kwargs.pop("backoff_factor", 1.0))
 
-        # Unicode data can't be sent over the wire. Convert it to UTF-8.
-        if "data" in kwargs and isinstance(kwargs["data"], str):
-            kwdata: str = kwargs["data"]
-            kwargs["data"] = kwdata.encode("utf8")
-
         # Set a user-agent if not already present
         version = (
             manager.__version__
             if manager.__version__
             else cls.DEFAULT_USER_AGENT_VERSION
         )
-        ua_header = {"User-Agent": f"Palace Manager/{version}"}
-        headers = ua_header | (kwargs.get("headers") or {})
-
-        # Make sure headers are encoded as utf-8
-        kwargs["headers"] = {
-            k.encode() if isinstance(k, str) else k: (
-                v.encode() if isinstance(v, str) else v
-            )
-            for k, v in headers.items()
-        }
```

Comment:
Again, requests is happy to accept headers as `str`, so no need to encode them to `bytes` ourselves.